mm2src/kdf_walletconnect/src/inbound_message.rs

```diff
@@ -61,31 +60,31 @@ pub(crate) async fn process_inbound_request(
 /// Processes an inbound WalletConnect response and sends the result to the provided message channel.
 ///
 /// Handles successful responses, errors, and specific session proposal processing.
-pub(crate) async fn process_inbound_response(
-    ctx: &WalletConnectCtxImpl,
-    response: Response,
-    topic: &Topic,
-    mut message_tx: UnboundedSender<SessionMessageType>,
-) {
+pub(crate) async fn process_inbound_response(ctx: &WalletConnectCtxImpl, response: Response, topic: &Topic) {
     let message_id = response.id();
-    let result = match response {
-        Response::Success(value) => match serde_json::from_value::<ResponseParamsSuccess>(value.result) {
-            Ok(data) => {
-                // TODO: maybe move to [send_proposal_request] and spawn in a different thread
-                if let ResponseParamsSuccess::SessionPropose(propose) = &data {
-                    process_session_propose_response(ctx, topic, propose).await.error_log();
-                }
-
-                Ok(SessionMessage {
+    let result = match &response {
+        Response::Success(value) => {
+            let data = serde_json::from_value::<ResponseParamsSuccess>(value.result.clone());
+            if let Ok(ResponseParamsSuccess::SessionPropose(propose)) = &data {
+                let mut pending_requests = ctx.pending_requests.lock().await;
+                pending_requests.remove(&message_id);
+                let _ = process_session_propose_response(ctx, topic, propose).await;
+                return;
+            };
+            data.map_to_mm(|err| WalletConnectError::SerdeError(err.to_string()))
+                .map(|data| SessionMessage {
                     message_id,
                     topic: topic.clone(),
                     data,
                 })
```

Comment:
i think having this as a match would simplify the structure. we also need a comment describing why we steal that pending request for session propose specifically.

a suggestion:
```rust
            match serde_json::from_value(value.result.clone()) {
                // If this is a session propose response, process it right away and return.
                // Session proposal responses are not waited for since it might take a long time
                // for the proposal to be accepted (user interaction). So they are handled in async fashion.
                Ok(ResponseParamsSuccess::SessionPropose(propose)) => {
                    let mut pending_requests = ctx.pending_requests.lock().await;
                    pending_requests.remove(&message_id);
                    let _ = process_session_propose_response(ctx, topic, &propose).await;
                    return;
                },
                Ok(response) => Ok(SessionMessage {
                    message_id,
                    topic: topic.clone(),
                    data: response,
                }),
                Err(e) => MmError::err(WalletConnectError::SerdeError(e.to_string())),
            }
```

another solution is to spawn a handler task for the session proposal that awaits for the response an handles it. that might be cleaner overall.

Reply:
we should also do something with the result of `process_session_propose_response`